mongodump/mongodump.go

```diff
@@ -511,15 +511,20 @@ func (dump *MongoDump) DumpIntent(intent *intents.Intent, buffer resettableOutpu
 	if err != nil {
 		return err
 	}
-	findQuery := &db.DeferredQuery{Coll: session.Database(intent.DB).Collection(intent.C)}
+	intendedDB := session.Database(intent.DB)
+	coll := intendedDB.Collection(intent.C)
+	findQuery := &db.DeferredQuery{Coll: coll}
 	switch {
 	case len(dump.query) > 0:
 		findQuery.Filter = dump.query
-	case dump.OutputOptions.ViewsAsCollections || dump.InputOptions.TableScan || intent.IsSpecialCollection() || intent.IsOplog():
+	case dump.OutputOptions.ViewsAsCollections || dump.InputOptions.TableScan || intent.IsSpecialCollection() ||
+		intent.IsOplog() || db.IsWiredTiger(intendedDB, intent.C):
```

Comment:
Do we need to check the storage engine for every collection, or can we just do it once at the beginning for the entire cluster?

Reply:
Just at the beginning.